src/mcp_server/tools/korrel8r_tools.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Dict, List, Optional
+import json
+import logging
+
+from common.pylogger import get_python_logger
+from core.correlation_service import CorrelationService
+from core.korrel8r_client import Korrel8rClient
+from core.korrel8r_service import fetch_goal_query_objects
+from core.config import (
+    KORREL8R_ENABLED,
+)
+from core.response_utils import make_mcp_text_response
+from mcp_server.exceptions import MCPException, MCPErrorCode
+
+
+logger = get_python_logger()
+# korrel8r_build_links tool removed per request
+
+
+def korrel8r_query_objects(query: str) -> List[Dict[str, Any]]:
+    """Execute a Korrel8r domain query and return objects.
+
+    Example query strings (see docs [korrel8r#_query_8](https://korrel8r.github.io/korrel8r/#_query_8)):
+      - alert:alert:{"alertname":"PodDisruptionBudgetAtLimit"}
+      - k8s:Pod:{"namespace", "llm-serving", "name":"vllm-inference-*"}
+      - loki:log:{"kubernetes.namespace_name":"llm-serving","kubernetes.pod_name":"p-abc"}
+      - trace:span:{".k8s.namespace.name":"llm-serving"}
+    """
+    if not KORREL8R_ENABLED:
```

Comment:
do you still need this check considering these tools will be called when `korrel8r` enabled ?

Reply:
not sure, how the tool will be called when it's not registered ?

Reply:
as we discussed in the call, lets keep this checking for now.